### PR TITLE
fix Riding Speed increases Druid Flight Forms

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:
-   fix: Item Carrot on a stick
-   fix: Enchant Gloves - Riding Skill
-   fix: Enchant Mithril Spurs

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16235

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Samsequel** [comment]
*testing in progress*

**Comment by Tereneckla** [comment]
Yes, enchant, spurs and carrot should stack

**Comment by Samsequel** [comment]
> Yes, enchant, spurs and carrot should stack

corrected the above then <3

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 1]
```diff
@@ -4553,6 +4553,20 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->Effects[EFFECT_0].TriggerSpell = 62585; // Mulgore Hatchling (fear)
     });
 
+    // item: Riding Crop/Skybreaker Whip (https://www.wowhead.com/tbc/spell=48776/mount-speed)
+    // item: Mithril Spurs (https://www.wowhead.com/tbc/spell=7215/mithril-spurs)
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellInfoCorrections.cpp:4557]: not how the comments work here.

Spellname only

**Comment by Kitzunu** [comment]
Conflict

**Comment by sudlud** [comment]
Can you please resolve the conflicts again

**Comment by sudlud** [comment]
So I'm using Carbonite right now 
https://felbite.com/addon/4846-carbonite/
to get a flight speed info within the client and tried to test this PR.

level 70 tauren druid in "swift flight form"
Riding Crop
.additem 25653

unequipped
speed: 280%

equipped
speed: 318%

If I understand correctly, the speed should be 280% even when "Riding Crop" is equipped?
Right now to me it looks like this PR does not have the desired effect, can you please check or give a hint what's going wrong on my side.

**Comment by grimgravy** [comment]
> Thanks for the update!
> 
> I'm using Carbonite right now https://felbite.com/addon/4846-carbonite/ to get a flight speed info within the client and tried to test this PR.
> 
> level 70 tauren druid in "swift flight form" Riding Crop (features spell 48776) .additem 25653
> 
> unequipped speed: 280%
> 
> equipped speed: 318%
> 
> If I understand correctly, the speed should be 280% even when "Riding Crop" is equipped?
> 
> Right now to me it looks like this PR does not have the desired effect, can you please check or give a hint what's going wrong on my side?
> 
> Thanks!

The Riding Crop is working with or without changes. However, the carrot on a stick and mithril spurs need modification

**Comment by sudlud** [comment]
So what's the state of this PR now? Read to be tested again?

**Comment by Kitzunu** [comment]
Conflicts

**Comment by alecsci** [comment]
What's the status for this ? if it's not ready can we label it differently ?

**Comment by grimgravy** [comment]
the effect is affecting other flying assemblies. i don't know what to do

**Comment by sudlud** [comment]
There have been changes made in this PR forth and back but there's no real "this is it - please review and test" state in this PR.

I would suggest closing it for now. Please feel free to open a new PR if you've found a viable solution for this issue.

**Comment by sogladev** [comment]
When comparing Mithril Spurs 59916 (no bug) with Riding Crop 48383, the DBC effects differ which explains the bug. Imo the DBC is wrong, or the below is using the wrong effect, but I can't find one that gives 10% speed with effect `SPELL_AURA_MOD_MOUNTED...SPEED`

https://github.com/azerothcore/azerothcore-wotlk/blob/bb0d292e1c2bb5619fbc3105e7cd919b45268748/src/server/scripts/Spells/spell_item.cpp#L228

```
items like carrot, mithril spurs
SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS
SPELL_AURA_MOD_MOUNTED_SPEED_ALWAYS

non-stackable bonus like crusader aura and riding crop
SPELL_AURA_MOD_FLIGHT_SPEED_NOT_STACK
SPELL_AURA_MOD_MOUNTED_SPEED_NOT_STACK
```

relevant core, handling looks correct

https://github.com/azerothcore/azerothcore-wotlk/blob/bb0d292e1c2bb5619fbc3105e7cd919b45268748/src/server/game/Entities/Unit/Unit.cpp#L14276

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -14279,7 +14279,10 @@ void Unit::UpdateSpeed(UnitMoveType mtype, bool forced)
                     stack_bonus     = GetTotalAuraMultiplier(SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS);
                 }
                 else             // Use not mount (shapeshift for example) auras (should stack)
+                {
                     main_speed_mod  = GetTotalAuraModifier(SPELL_AURA_MOD_INCREASE_FLIGHT_SPEED) + GetTotalAuraModifier(SPELL_AURA_MOD_INCREASE_VEHICLE_FLIGHT_SPEED);
+                    stack_bonus = GetTotalAuraMultiplier(SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS);
```
Comment (sogladev) [inline src/server/game/Entities/Unit/Unit.cpp:14284]: This change is not needed; This adds mount speed increases (`SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS`) to affect non-mounts (`SPELL_AURA_MOD_INCREASE_VEHICLE_FLIGHT_SPEED`)

e.g. carrot and spurs will now boost flight form

Comment (sogladev) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: Modifying the dbc of 48383 to apply below auras should work
```
SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS
SPELL_AURA_MOD_MOUNTED_SPEED_ALWAYS
```

Comment (grimgravy) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: > This change is not needed; This adds mount speed increases (`SPELL_AURA_MOD_MOUNTED_FLIGHT_SPEED_ALWAYS`) to affect non-mounts (`SPELL_AURA_MOD_INCREASE_VEHICLE_FLIGHT_SPEED`)
> 
> e.g. carrot and spurs will now boost flight form

application is correct

Comment (sudlud) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: So this should be updated in spell info correction?

Comment (sogladev) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: y below should work. So it behaves more like mithril spurs 

`SPELL_RIDING_CROP_EFFECT = 48383,` from spell_item.cpp
```
diff --git a/src/server/game/Spells/SpellInfoCorrections.cpp b/src/server/game/Spells/SpellInfoCorrections.cpp
index ca47ac9fe1..66e671e7fb 100644
--- a/src/server/game/Spells/SpellInfoCorrections.cpp
+++ b/src/server/game/Spells/SpellInfoCorrections.cpp
@@ -4852,2 +4852,9 @@ void SpellMgr::LoadSpellInfoCorrections()
 
+    // Mount Speed (Riding Crop effect)
+    ApplySpellFix({ 48383 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->Effects[EFFECT_0].ApplyAuraName = SPELL_AURA_MOD_MOUNTED_SPEED_ALWAYS;
+        spellInfo->Effects[EFFECT_1].ApplyAuraName = SPELL_AURA_MOD_FLIGHT_SPEED_NOT_STACK;
+    });
+
```

spellwork
riding_crop_spurs_dbc

Comment (grimgravy) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: I understand, but it doesn't solve the problem, because it's not in that spell

Comment (grimgravy) [reply src/server/game/Entities/Unit/Unit.cpp:14284]: > So this should be updated in spell info correction?

there was a problem, the change I made is stack with crusader aura

**Comment by sudlud** [comment]
just repeated my test of https://github.com/azerothcore/azerothcore-wotlk/pull/16761#issuecomment-2002599612 with the same result, so the current state of the PR is still not solving the issue as it seems to me.

**Comment by sudlud** [comment]
Thanks again for working on this PR - but this can probably be closed in favor of #20082 which provides a working solution for the issue.

**Comment by sudlud** [comment]
Closing as #20079 has been merged now.

**Pull request closed without merging**